src/tiramisu_core.cpp

```diff
@@ -7810,4 +7810,231 @@ void tiramisu::buffer::tag_gpu_register() {
     set_auto_allocate(false);
 }
 
+
+/**
```

Comment:
No need to write a comment on the source code. Usually documentation of how a function should be used should be in the header file while documentation related to the implementation details only (i.e., how the function works) should be in the source file.